cni/pkg/nodeagent/net.go

```diff
@@ -277,6 +277,12 @@ func (s *NetServer) DelPodFromMesh(ctx context.Context, pod *corev1.Pod) error {
 	return nil
 }
 
+// syncHostIPSets is called after the host node ipset has been created (or found + flushed)
+// during initial snapshot creation, it will insert every snapshotted pod's IP into the set.
+//
+// The set does not allow dupes (obviously, that would be undefined) - but in the real world due to misconfigured
+// IPAM or other things, we may see two pods with the same IP on the same node - we will skip the dupes,
```

Comment:
I am not sure this logic is quite right, broadly. AFAIK it is illegal for 2 pods to be _running_ with the same IP on the same node. Its not illegal to have 2 pods on the same with the same IP though and its very easy to get into this state -- just run >255 pods on a node that Complete. They stick around forever, including the node and IP.

So we should probably filter out Completed pods basically

Reply:
Oh, that is a good hint. Because we observed this problem (https://github.com/istio/istio/pull/51081) usually in conjunction with a kubernetes `CronJob` which ran around the time period when the "duplicated" IP addresses popped up.

And also in this PR (https://github.com/istio/istio/pull/51429) a kubernetes `CronJob` ran around the CNI upgrade from `1.22.0` to `1.22.1`

Reply:
> They stick around forever, including the node and IP.

ugh. and this would have been filtered out of spurious adds because we check readiness elsewhere.

Reply:
this creates a bit of a problem for the informer generally, as AFAICT we won't get an informer event on those pods reaching a completion status.

Reply:
nevermind, we do.